### PR TITLE
Kill process instead of container, on container restart. 

### LINKED ISSUE #2123: restarting containers get killed if attachment fails
When executing `docker run` or `docker start` against the proxy, and a failure occurs when attaching the container to the weave network, we want the run/start to fail. We do this by `docker kill`ing the container - the `--or-die` variant of `weave attach`. So far so good...

But the very same code is used for re-attaching a restarting container. And `docker kill`ing a restarting container will prevent it from ever restarting. Which is (probably?) wrong.

The same "or die" behaviour is exhibited by `weave restart`.

I wonder whether killing the *process* might have the desired effect of returning an error to the CLI, while keeping docker's restart logic operational.

### PR DESCRIPTION
**Pull request by bboreham**
Fixes #2123 

I shall try to add a smoke-test.

### CODE REVIEW

File: proxy/proxy.go [hunk 1]
```diff
@@ -498,7 +501,11 @@ func (proxy *Proxy) attach(containerID string, orDie bool) error {
 		args = append(args, "--no-multicast-route")
 	}
 	if orDie {
-		args = append(args, "--or-die")
+		if killContainer {
+			args = append(args, "--or-die")
+		} else {
+			args = append(args, "--or-kill")
+		}
```
Comment (rade) [inline proxy/proxy.go]: This is either the wrong way round or the naming is really confusing.

Comment (bboreham) [inline proxy/proxy.go]: `--or-die` is the existing behaviour; it kills the container.
The new behaviour is to kill the process.
Maybe `--or-kill-process` ?

Comment (bboreham) [inline proxy/proxy.go]: amended boolean arg to `killProcess`

**Comment by rade** [comment]
> I shall try to add a smoke-test.

yes, please.

**Comment by bboreham** [comment]
Smoke-test added

**Pull request merged**